src/js/popper.js

```diff
@@ -0,0 +1,1210 @@
+      (e.left += i * r),
+      (e.right += i * r),
+      e
+    );
+  }
+  function m(e, t) {
+    var o = "x" === t ? "Left" : "Top",
+      n = "Left" == o ? "Right" : "Bottom";
+    return (
+      parseFloat(e["border" + o + "Width"], 10) +
+      parseFloat(e["border" + n + "Width"], 10)
+    );
+  }
+  function h(e, t, o, n) {
+    return ee(
+      t["offset" + e],
+      t["scroll" + e],
+      o["client" + e],
+      o["offset" + e],
+      o["scroll" + e],
+      r(10)
+        ? parseInt(o["offset" + e]) +
+            parseInt(n["margin" + ("Height" === e ? "Top" : "Left")]) +
+            parseInt(n["margin" + ("Height" === e ? "Bottom" : "Right")])
+        : 0
+    );
+  }
+  function c(e) {
+    var t = e.body,
+      o = e.documentElement,
+      n = r(10) && getComputedStyle(o);
+    return { height: h("Height", t, o, n), width: h("Width", t, o, n) };
+  }
+  function g(e) {
+    return le({}, e, { right: e.left + e.width, bottom: e.top + e.height });
+  }
+  function u(e) {
+    var o = {};
+    try {
+      if (r(10)) {
+        o = e.getBoundingClientRect();
+        var n = l(e, "top"),
+          i = l(e, "left");
+        (o.top += n), (o.left += i), (o.bottom += n), (o.right += i);
+      } else o = e.getBoundingClientRect();
+    } catch (t) {}
+    var p = {
+        left: o.left,
+        top: o.top,
+        width: o.right - o.left,
+        height: o.bottom - o.top,
+      },
+      s = "HTML" === e.nodeName ? c(e.ownerDocument) : {},
+      d = s.width || e.clientWidth || p.width,
+      a = s.height || e.clientHeight || p.height,
+      f = e.offsetWidth - d,
+      h = e.offsetHeight - a;
+    if (f || h) {
+      var u = t(e);
+      (f -= m(u, "x")), (h -= m(u, "y")), (p.width -= f), (p.height -= h);
+    }
+    return g(p);
+  }
+  function b(e, o) {
+    var i = 2 < arguments.length && void 0 !== arguments[2] && arguments[2],
+      p = r(10),
+      s = "HTML" === o.nodeName,
+      d = u(e),
+      a = u(o),
+      l = n(e),
+      m = t(o),
+      h = parseFloat(m.borderTopWidth, 10),
+      c = parseFloat(m.borderLeftWidth, 10);
+    i && s && ((a.top = ee(a.top, 0)), (a.left = ee(a.left, 0)));
+    var b = g({
+      top: d.top - a.top - h,
+      left: d.left - a.left - c,
+      width: d.width,
+      height: d.height,
+    });
+    if (((b.marginTop = 0), (b.marginLeft = 0), !p && s)) {
+      var w = parseFloat(m.marginTop, 10),
+        y = parseFloat(m.marginLeft, 10);
+      (b.top -= h - w),
+        (b.bottom -= h - w),
+        (b.left -= c - y),
+        (b.right -= c - y),
+        (b.marginTop = w),
+        (b.marginLeft = y);
+    }
+    return (
+      (p && !i ? o.contains(l) : o === l && "BODY" !== l.nodeName) &&
+        (b = f(b, o)),
+      b
+    );
+  }
+  function w(e) {
+    var t = 1 < arguments.length && void 0 !== arguments[1] && arguments[1],
+      o = e.ownerDocument.documentElement,
+      n = b(e, o),
+      i = ee(o.clientWidth, window.innerWidth || 0),
```

Comment:
Codacy found an issue: ['ee' was used before it was defined.](https://app.codacy.com/manual/hello_100/jca-ui-kit/pullRequest?prid=5731813)